README.unittests.rst

```diff
@@ -216,6 +217,75 @@ Additional steps specific to individual databases are as follows::
 
      ALTER DATABASE MyDatabase SET READ_COMMITTED_SNAPSHOT ON
 
+Docker Configurations
+---------------------
+
+The SQLAlchemy test can run against database running in Docker containers.
+This ensures that they are empty and that their configuration is not influenced
+by any local usage.
+
+The following configurations are just examples that developers can use to
+quickly set up a local environment for SQLAlchemy development. They are **NOT**
+intended for production use!
+
+**PostgreSQL configuration**::
+
+    # only needed if a local image of postgres is not already present
+    docker pull postgres:12
+
+    # create the container with the proper configuration for sqlalchemy
+    docker run --rm -e POSTGRES_USER='scott' -e POSTGRES_PASSWORD='tiger' -e POSTGRES_DB='test' -p 5432:5432 -d --name postgres postgres:12-alpine
+
+    # configure the database
+    sleep 10
+    docker exec -ti postgres psql -U scott -c 'CREATE SCHEMA test_schema;' test
+    docker exec -ti postgres psql -U scott -c 'CREATE SCHEMA test_schema_2;' test
+    # this last command is optional
+    docker exec -ti postgres bash sed -i 's/#max_prepared_transactions = 0/max_prepared_transactions = 10/g' /var/lib/postgresql/data/postgresql.conf
+
+    # To stop the container. It will also remove it.
+    docker stop postgres
+
+**MySQL configuration**::
+
+    # only needed if a local image of mysql is not already present
+    docker pull mysql:8
+
+    # create the container with the proper configuration for sqlalchemy
+    docker run --rm -e MYSQL_USER='scott' -e MYSQL_PASSWORD='tiger' -e MYSQL_DATABASE='test' -e MYSQL_ROOT_PASSWORD='password' -p 3306:3306 -d --name mysql mysql:8
+
+    # configure the database
+    sleep 20
+    docker exec -ti mysql mysql -u root -ppassword -D test -w -e 'GRANT ALL PRIVILEGES ON *.* TO scott'
+    docker exec -ti mysql mysql -u root -ppassword -D test -w -e 'CREATE DATABASE test_schema'
+
+    # To stop the container. It will also remove it.
+    docker stop mysql
+
+**MSSQL configuration**::
+
+    # only needed if a local image of mssql is not already present
+    docker pull mcr.microsoft.com/mssql/server:2019-CU1-ubuntu-16.04
+
+    # create the container with the proper configuration for sqlalchemy
+    # it will use the Developer version
+    docker run --rm -e 'ACCEPT_EULA=Y' -e 'SA_PASSWORD=yourStrong(!)Password' -p 1433:1433 -d --name mssql mcr.microsoft.com/mssql/server:2019-CU1-ubuntu-16.04
```

Comment:
**mike bayer** (zzzeek) wrote:

does this bind to the host's network so that the 127.0.0.1:1433 works ?   I thought you needed --net=host or is that docker's default ?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1764

Reply:
**Federico Caselli** (CaselIT) wrote:

In the default configuration it binds to the host network.
But I just noticed that I missed the 127.0.0.1. By default docker binds to 0.0.0.0 on the host and we do not want that!

I'll fix it also on other cases

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1764